res/controllers/Numark-Scratch-scripts.js

```diff
@@ -0,0 +1,417 @@
+// eslint-disable-next-line no-var
+var NumarkScratch = {};
+
+/*
+ * USER CONFIGURABLE SETTINGS.
+ */
+
+// Defines the Beatloop Roll sizes for the 4 pads, for available values see:
+// https://manual.mixxx.org/2.4/en/chapters/appendix/mixxx_controls.html#control-[ChannelN]-beatlooproll_X_activate
+// Default: [0.25, 0.5, 1, 2]
+NumarkScratch.autoLoopSizes = [
+    "0.25",
+    "0.5",
+    "1",
+    "2",
+];
+
+// Defines how the Loop Encoder functions.
+// If 'true' the Encoder scrolls the library/loads track. Shift + Encoder manages looping.
+// If 'false' the Encoder manages looping. Shift + Encoder scrolls the library/loads track (Serato default).
+// Default: false
+NumarkScratch.invertLoopEncoderFunction = false;
+
+// Defines the bightness of button LEDs when they inactive.
+// '0x00' sets the LEDSs to completely off. '0x01 sets the LEDs to dim.
+// Default: 0x01 (dim)
+NumarkScratch.LOW_LIGHT = 0x01;
+
+/*
+ * CODE
+ */
+
+
+components.Button.prototype.off = NumarkScratch.LOW_LIGHT;
+
+NumarkScratch.init = function() {
+    // Initialize component containers
+    NumarkScratch.deck = new components.ComponentContainer();
+    NumarkScratch.effect = new components.ComponentContainer();
+    for (let i = 0; i < 2; i++) {
+        NumarkScratch.deck[i] = new NumarkScratch.Deck(i + 1);
+        NumarkScratch.effect[i] = new NumarkScratch.EffectUnit(i + 1);
+    }
+
+    NumarkScratch.xfader = new NumarkScratch.XfaderContainer();
+
+    const createVuCallback = function(deckOffset) {
+        return function(value) {
+            midi.sendShortMsg(0xB0 + deckOffset, 0x1F, value * 90);
+        };
+    };
+
+    engine.makeConnection("[Channel1]", "vu_meter", createVuCallback(0));
+    engine.makeConnection("[Channel2]", "vu_meter", createVuCallback(1));
+
+    // Trigger is needed to initialize lights to 0x01
+    NumarkScratch.deck.forEachComponent(function(component) {
+        component.trigger();
+    });
+    NumarkScratch.effect.forEachComponent(function(component) {
+        component.trigger();
+    });
```

Comment:
if you don't need to access `this` (at least in a component) you can write this a little nicer using arrow functions (this doesn't work for "methods" of componentJS though):

```suggestion
    const createVuCallback = deckOffset => value => midi.sendShortMsg(0xB0 + deckOffset, 0x1F, value * 90);

    engine.makeConnection("[Channel1]", "vu_meter", createVuCallback(0));
    engine.makeConnection("[Channel2]", "vu_meter", createVuCallback(1));

    // Trigger is needed to initialize lights
    NumarkScratch.deck.forEachComponent(component => component.trigger());
    NumarkScratch.effect.forEachComponent(component => component.trigger());
```

Reply:
learn something new. done